### PR TITLE
AbstractSqlDataService refactoring and code reformatting

### PR DESCRIPTION
**Pull request by washibana**

### CODE REVIEW

File: backend/Origam.DA.Service/AbstractSqlDataService.cs [hunk 1]
```diff
@@ -478,174 +503,186 @@ internal override IDbTransaction GetTransaction(string transactionId, IsolationL
...
+	                        recordId, transactionId);
                     }
                 }
 				if(newTransaction)
 				{
 					ResourceMonitor.Commit(transactionId);
 				}
 			}
-			catch(DBConcurrencyException e)
+			catch(DBConcurrencyException ex)
```
Comment (JindrichSusen) [inline backend/Origam.DA.Service/AbstractSqlDataService.cs:643]: Before you changed variable name "ex" to "exception", now you change "e" to "ex"? I think that "ex" is just fine as name of a chatch variable, just like "i" is ok in a for loop variable. I suggest we use "ex" every where.

File: backend/Origam.DA.Service/AbstractSqlDataService.cs [hunk 2]
```diff
@@ -1723,400 +1844,531 @@ private List<ColumnData> GetAllQueryColumns(DataStructureQuery query)
...
+					                var foundPair = false;
+					                for(var i = 1; i < 17; i++)
+					                {
+						                if((rows[0]["cKeyCol" + i] 
+						                    != DBNull.Value) 
+						                   && ((string)rows[0]["cKeyCol" + i] 
+						                       == foreignKey) 
+						                   && ((string)rows[0]["cRefCol" + i] 
+						                       == primaryKey))
+						                {
```
Comment (JindrichSusen) [inline backend/Origam.DA.Service/AbstractSqlDataService.cs:2116]: I don't think breaking the statement into 3 more lines made it any more readable. The real problem here is the nesting. Is the result under 80 characters wide anyway?

File: backend/Origam.DA.Service/AbstractSqlDataService.cs [hunk 3]
```diff
@@ -1723,400 +1844,531 @@ private List<ColumnData> GetAllQueryColumns(DataStructureQuery query)
...
+			                = (differenceDescription == "" ? "" : "; ")
+			                  + "DataType: Schema-"
+			                  + abstractSqlCommandGenerator.DdlDataType(
+				                  fieldMappingItem.DataType, 
+				                  fieldMappingItem.MappedDataType)
+			                  + ", Database-" + (string)row["DATA_TYPE"];
+	                }
+	                if((fieldMappingItem.DataType == OrigamDataType.String)
+	                   && !row.IsNull("CHARACTER_MAXIMUM_LENGTH")
+	                   && ((int)row["CHARACTER_MAXIMUM_LENGTH"]
```
Comment (JindrichSusen) [inline backend/Origam.DA.Service/AbstractSqlDataService.cs:2318]: I think It looks more readbale to add an indentation before "&&" like you did here. But this seems to be an exception there are many if statements with the indents befere the new line missing.

**Review by JindrichSusen — CHANGES_REQUESTED** [review]

File: backend/Origam.DA.Service/AbstractSqlDataService.cs [hunk 4]
```diff
@@ -431,18 +440,29 @@ internal override IDbTransaction GetTransaction(string transactionId, IsolationL
...
+			}
+			if(entity.RelationType != RelationType.Normal)
+			{
+				return false;
+			}
+			if(!(rootEntity.Entity is TableMappingItem mappingItem))
+			{
+				return false;
+			}
+            var rootEntityTableName = mappingItem.MappedObjectName;
```
[the remark this answers is not in the document]
Comment (washibana) [reply backend/Origam.DA.Service/AbstractSqlDataService.cs:460]: Indentation is in place, but github is interpreting is in its own way. It's most likely related to the type of whitespace.

File: backend/Origam.DA.Service/AbstractSqlDataService.cs [hunk 5]
```diff
@@ -61,123 +62,129 @@ internal class DataLoader
...
 					{
 						connection.Close();
 					}
-				} 
-				catch{}
-				if(this.Transaction == null)
+				}
+				catch
+				{
+					// ignored
```
[the remark this answers is not in the document]
Comment (washibana) [reply backend/Origam.DA.Service/AbstractSqlDataService.cs:151]: Removed the comments.

File: backend/Origam.DA.Service/AbstractSqlDataService.cs [hunk 6]
```diff
@@ -478,174 +503,186 @@ internal override IDbTransaction GetTransaction(string transactionId, IsolationL
...
+            var deletedRowIds = new ArrayList();
+			var rowStates = new[]
+			{
+				DataRowState.Deleted, DataRowState.Added, DataRowState.Modified
+			};
 			DataTable changedTable = null;
 			try
 			{
-				foreach(DataRowState rowState in rowStates)
+				foreach(var rowState in rowStates)
```
[the remark this answers is not in the document]
Comment (washibana) [reply backend/Origam.DA.Service/AbstractSqlDataService.cs:531]: I believe that in marked case usage of `var` is along your description. I agree with cases marked as incorrect.

File: backend/Origam.DA.Service/AbstractSqlDataService.cs (hunk 1 above)
Comment (washibana) [reply backend/Origam.DA.Service/AbstractSqlDataService.cs:643]: The applied rule is tu use `ex` in `catch` and to use `exception` as argument of the function.

File: backend/Origam.DA.Service/AbstractSqlDataService.cs (hunk 2 above)
Comment (washibana) [reply backend/Origam.DA.Service/AbstractSqlDataService.cs:2116]: Result was under 80 characters. I've extracted the part of the code to another function to remove nesting.

File: backend/Origam.DA.Service/AbstractSqlDataService.cs (hunk 3 above)
Comment (washibana) [reply backend/Origam.DA.Service/AbstractSqlDataService.cs:2318]: The rest of the code is adjusted accordingly.

File: backend/Origam.DA.Service/AbstractSqlDataService.cs (hunk 1 above)
Comment (JindrichSusen) [reply backend/Origam.DA.Service/AbstractSqlDataService.cs:643]: ok

**Review by JindrichSusen — APPROVED** [review]

File: backend/Origam.Workflow/ProfilingTools.cs [hunk 7]
```diff
@@ -115,15 +115,15 @@ bool FuncToExecute()
...
+            ThreadContext.Properties["currentTaskPath"] = null;
             ThreadContext.Properties["currentTaskId"] = null;
             ThreadContext.Properties["ServiceMethodName"] = null;
         }
 
         public static void SetCurrentTaskToThreadLoggingContext(
             ServiceMethodCallTask task)
         {
-            ThreadContext.Properties["currentTaskpath"] = task.Path;
+            ThreadContext.Properties["currentTaskPath"] = task.Path;
```
[the remark this answers is not in the document]
Comment (washibana) [reply backend/Origam.Workflow/ProfilingTools.cs:126]: All found instances were adjusted. They were only in `AbstractSqlDataService`.

**Comment by tvavrda** [comment]
Approved. I think constants should be used instead of strings here though: `ThreadContext.Properties["currentTaskPath"] = task.Path;`

**Pull request merged**